Raise an error if batch transfer hooks are overridden with IPUAccelerator (#13961)
Lightning-AI/lightning: docs/source-pytorch/accelerators/gpu_intermediate.rst

    :doc:`Manual Optimization <../model/manual_optimization>` with DP. Use DDP which is more stable and at least 3x faster.

.. warning:: DP only supports scattering and gathering primitive collections of tensors like lists, dicts, etc.

>     Therefore the hooks :meth:`~pytorch_lightning.core.hooks.ModelHooks.on_before_batch_transfer`,
>     :meth:`~pytorch_lightning.core.hooks.ModelHooks.transfer_batch_to_device`, :meth:`~pytorch_lightning.core.hooks.ModelHooks.on_after_batch_transfer`
>     does not apply in this mode and if you have overridden it, it will raise an exception.

Reviewer: ```suggestion
    Therefore the hooks :meth:`~pytorch_lightning.core.hooks.ModelHooks.on_before_batch_transfer`,
    :meth:`~pytorch_lightning.core.hooks.ModelHooks.transfer_batch_to_device` and :meth:`~pytorch_lightning.core.hooks.ModelHooks.on_after_batch_transfer`
    do not apply in this mode and if you have overridden any of them, an exception will be raised.
```
changing to the plural also requires some grammatical changes.
